src/runtime_src/tools/xclbinutil/Section.cxx

```diff
@@ -73,85 +140,108 @@ Section::setName(const std::string &_sSectionName)
 
 void
 Section::getKinds(std::vector< std::string > & kinds) {
-  for (auto & item : m_mapNameToId) {
-    kinds.push_back(item.first);
-  }
+  for (auto & item : getSectionTypes()) 
+    kinds.push_back(item->getSectionName());
 }
```

Comment:
I think this function should return a std::vector

Reply:
Agree.  Refactored.